### PR TITLE
HHH-17559 - Prefer Java Time handling for all temporal values

### PR DESCRIPTION
**Pull request by sebersole**
https://hibernate.atlassian.net/browse/HHH-17575
https://hibernate.atlassian.net/browse/HHH-17559

### CODE REVIEW

**Comment by sebersole** [comment]
This first commit implements "phase 1" for Instant, LocalDateTime, LocalTime and LocalDate.

1. Create JdbcType variants for each Java Time contract (LocalDateJdbcType, …) with corresponding SqlTypes entries (LOCAL_DATE, …).  These would use the JDBC get/setObject calls.

File: hibernate-core/src/test/java/org/hibernate/orm/test/temporal/FractionalSecondsTests.java [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+			final TestEntity3 testEntity = session.find( TestEntity3.class, 1 );
+			assertThat( testEntity.theInstant ).isEqualTo( DateTimeUtils.roundToSecondPrecision( start, 3 ) );
+		} );
+	}
+
+	@Test
+	@DomainModel(annotatedClasses = TestEntity9.class)
+	@SessionFactory
+	@SkipForDialect( dialectClass = MariaDBDialect.class, reason = "MariaDB only supports precision <= 6" )
+	@SkipForDialect( dialectClass = MySQLDialect.class, reason = "MySQL only supports precision <= 6" )
```
Comment (beikov) [inline hibernate-core/src/test/java/org/hibernate/orm/test/temporal/FractionalSecondsTests.java:181]:
```suggestion
	@SkipForDialect( dialectClass = MySQLDialect.class, matchSubTypes = true, reason = "MySQL-like dialects only supports precision <= 6" )
```

File: hibernate-core/src/test/java/org/hibernate/orm/test/temporal/JavaTimeFractionalSecondsTests.java [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+			final TestEntity3 testEntity = session.find( TestEntity3.class, 1 );
+			assertThat( testEntity.theInstant ).isEqualTo( DateTimeUtils.roundToSecondPrecision( start, 3 ) );
+		} );
+	}
+
+	@Test
+	@DomainModel(annotatedClasses = TestEntity9.class)
+	@SessionFactory
+	@SkipForDialect( dialectClass = MariaDBDialect.class, reason = "MariaDB only supports precision <= 6" )
+	@SkipForDialect( dialectClass = MySQLDialect.class, reason = "MySQL only supports precision <= 6" )
```
Comment (beikov) [inline hibernate-core/src/test/java/org/hibernate/orm/test/temporal/JavaTimeFractionalSecondsTests.java:183]:
```suggestion
	@SkipForDialect( dialectClass = MySQLDialect.class, matchSubTypes = true, reason = "MySQL-like dialects only supports precision <= 6" )
```

File: hibernate-core/src/test/java/org/hibernate/orm/test/temporal/FractionalSecondsTests.java (hunk 1 above)
Comment (sebersole) [reply hibernate-core/src/test/java/org/hibernate/orm/test/temporal/FractionalSecondsTests.java:181]: I'd prefer to leave these as is.  I made that choice explicitly.

File: hibernate-core/src/test/java/org/hibernate/orm/test/temporal/JavaTimeFractionalSecondsTests.java (hunk 2 above)
Comment (sebersole) [reply hibernate-core/src/test/java/org/hibernate/orm/test/temporal/JavaTimeFractionalSecondsTests.java:183]: Same.

**Pull request merged**